docarray/array/storage/weaviate/backend.py

```diff
@@ -352,10 +352,9 @@ def _doc2weaviate_create_payload(self, value: 'Document'):
         :param value: document to create a payload for
         :return: the payload dictionary
         """
-        columns_dict = {key: val for [key, val] in self._config.columns}
         extra_columns = {
-            col: self._map_column(value.tags.get(col), columns_dict[col])
-            for col, _ in self._config.columns
+            k: self._map_column(value.tags.get(k), v)
+            for k, v in self._config.columns.items()
```

Comment:
Maybe we could give `k` and `v` descriptive names for readability? I think `k` -> `col`; `v` -> `col_type` should be correct?